### PR TITLE
Theming using theming package

### PR DESCRIPTION
**Pull request by PetrSnobelt**
This Pull Request meets the following criteria:

* [x] Tests have been added/adjusted for my new feature
* [x] New Components are registered in index.js of my project

Sample component which is used for storybook tests is not exported

### CODE REVIEW

File: src/Theming/ThemeSwitcher.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  children: any,
+  mainTheme?: typeof defaultTheme,
+  themes: any[],
+};
+
+type ThemeState = {
+  themeIndex: number,
+};
+
+export default class ThemeSwitherSample extends React.Component<ThemeProps, ThemeState> {
```
Comment (jaroslav-kubicek) [inline src/Theming/ThemeSwitcher.js]: I would suggest to have the same name for default export as for the filename

File: src/Theming/ThemeSwitcher.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+// @flow
+import * as React from "react";
+import defaultTheme from "orbit-design-token";
+
+import ThemeProvider from "./ThemeProvider";
+
+type ThemeProps = {
+  children: any,
```
Comment (jaroslav-kubicek) [inline src/Theming/ThemeSwitcher.js]: children is not definitely any :)

Comment (infi-pc) [reply src/Theming/ThemeSwitcher.js]: nice catch. just hint for better code reviews, if you know the solution it is good to give some hint. Something like:
Don't use `any` for children, React children have type `Node` see https://flow.org/en/docs/react/children/

```
import type { Node } from "react"
...
children: Node,
...
```

File: src/Theming/ThemeSwitcher.js (hunk 1 above)
Comment (PetrSnobelt) [reply src/Theming/ThemeSwitcher.js]: OK

File: src/Theming/ThemeSwitcher.js (hunk 2 above)
Comment (PetrSnobelt) [reply src/Theming/ThemeSwitcher.js]: OK, Fixed

**Comment by tomashapl** [comment]
@PetrSnobelt please can you strict define, `defaultTheme` object to use it in flow type annotation

File: src/Theming/ThemeSwitcher.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// @flow
+import * as React from "react";
+import defaultTheme from "orbit-design-token";
+
+import ThemeProvider from "./ThemeProvider";
+
+type Props = {
+  children: React.Node,
+  mainTheme?: typeof defaultTheme,
+  themes: any[],
```
[the remark this answers is not in the document]
Comment (PetrSnobelt) [reply src/Theming/ThemeSwitcher.js]: 👍

**Pull request merged**